### DISCUSSION TITLE
How to display all overloads of a function and step through them

### DISCUSSION THREAD

**Discussion by lime21312** [hrsh7th/nvim-cmp#995] [Q&A] [answered]
I may have missed this in the docs but is there any way to display and choose between all overloads of a function? As is it just shows overloaded functions like this `foo(...) [x overloads]`.

**Comment by jamestrew** [comment]
I think it would help if you'd be more specific. What language are you working in?
Something like that you're asking does work for me in cpp with pretty default settings.

![image](https://user-images.githubusercontent.com/66286082/169446237-e5192869-1d65-46b8-9ec5-6dc3b396531f.png)
Then selecting `func` gives me
![image](https://user-images.githubusercontent.com/66286082/169446266-8868cd37-b289-462d-9e19-8adac723790d.png)

**Reply by lime21312** [reply]
I'm working with cpp as well.

> Then selecting func gives me

Pressing enter and selecting the function doesn't do anything for me.

Here is how my lsp-config setup looks like:
   ```
 for _, lsp in ipairs(servers) do
     lspconfig[lsp].setup {
       -- on_attach = my_custom_on_attach,   
       capabilities = capabilities
     }                             
    end 
```

**Reply by jamestrew** [reply]
Hmm... make sure your `<CR>` key is bound properly like so `['<CR>'] = cmp.mapping.confirm({ select = true })` in the `setup` function.

**Comment by S4deghN** [comment, marked as answer]
You need to use this: [https://github.com/hrsh7th/cmp-nvim-lsp-signature-help](url)
`nvim-cmp` is a modular plugin and you need to install features you want manually.
If you already have set up the main plugin, adding this feature is as easy as installing the plugin the way you prefer and adding this to your config:
```lua

require'cmp'.setup {
  sources = {
    { name = 'nvim_lsp_signature_help' }
  }
}
```

**Reply by nikfp** [reply]
Understanding that this is a year old, the link provided in this answer is hitting a 404. Any chance for an update? In the meantime the [correct link is here](https://github.com/hrsh7th/cmp-nvim-lsp-signature-help) if needed. 

Thank you all for all you do!

**Comment by git-dimiz** [comment]
Hi,

I've kind of a similar question. Let's say I have these two functions:

```
void fun();
void fun(int i);
```

The LSP will show me that there are two overloads. But if the approach mentioned above I will not see the first version of `fun` because it has no arguments. Is there no way to cycle through the two overloads and select the appropriate one?
